app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java

```diff
@@ -435,12 +435,32 @@ protected void onResume() {
         setNotificationCount();
     }
 
+    /**
+     * This function will clear the cache made by the Nearby feature
+     */
+    public void clearNearbyCache() {
```

Comment:
Use the MVP pattern.

Reply:
Ok, I moved the code in NearbyParentFragment in NearbyController and I made the NearbyParentFragment's method only call the controller's.

I didn't change the MainActivity one since it just finds the right fragment to be able to call the function

Reply:
Instead of finding the fragment, you should access the singletone presenter object and implement your method there, ideally.

Reply:
There are some examples for how to do in MainActivity, you can check them.